File: 8 kyu/Gravity Flip/solution.rb
def flip(dir, boxes)
  dir == "R" ? boxes.sort : boxes.sort.reverse
end

____________________________
def flip(dir, boxes)
  boxes.sort_by(&(dir == 'R' ? :+@ : :-@))
end

____________________________
def flip(dir, boxes)
  eval("boxes.sort#{dir == 'L' ? '.reverse' : ''}")
end

____________________________
def flip(dir, boxes) # declare function taking dir and boxes arguments
  if dir == 'R' # test dir input against the entered input'R'
    boxes.sort # to represent the movement of highest magnitude columns to the right
  elsif dir == 'L' # elsif not else, because an input could be entered other than 'L'
    boxes.sort.reverse # to represent the movement of highest magnitude columns to the left
  end # end 'if elsif' statement
end # end function declaration

File: 8 kyu/Gravity Flip/solution.clj
(ns kata)

(defn flip [d a]
  (sort (case d \R < \L >) a))

_____________________________________________
(ns kata)

(defn flip [d a]
  (case d
    \R (sort < a)
    \L (sort > a)))

_____________________________________________
(ns kata)

(defn flip [gravity boxes]
  (def gravity-side (case gravity \R < \L >))
  (sort gravity-side boxes)
)

_____________________________________________
(ns kata)

(defn flip [d a]
  (if (= d \L) (sort > a) (sort a)))
